src/cache/server/cache.go

```diff
@@ -140,6 +140,23 @@ func (cache *Cache) removeAndDeleteFile(p string, file *cachedFile) {
 // return whatever's been stored there, which might be a directory and therefore contain
 // multiple files to be returned.
 func (cache *Cache) RetrieveArtifact(artPath string) (map[string][]byte, error) {
+	ret := map[string][]byte{}
+	if strings.Contains(artPath, "*") {
```

Comment:
good point, yeah. Have made it more robust - there are a pretty small set of special characters. I do think it's a good idea to do the check rather than globbing every time though.